lib/internal/modules/cjs/loader.js

```diff
@@ -1669,6 +1672,11 @@ Module._extensions['.js'] = function(module, filename) {
   // If already analyzed the source, then it will be cached.
   const content = getMaybeCachedSource(module, filename);
 
+  const isAllowedToExecute = ci.isAllowedToExecuteFile(filename);
```

Comment:
This might be better placed inside Module._load (search for permission.isEnabled() on this file).

Reply:
We shouldn't even call this function for non-windows environments.